apis/fluentbit/v1alpha2/clusterfluentbitconfig_types.go

```diff
@@ -118,6 +118,8 @@ type Service struct {
 	EmitterName        string `json:"emitterName,omitempty"`
 	EmitterMemBufLimit string `json:"emitterMemBufLimit,omitempty"`
 	EmitterStorageType string `json:"emitterStorageType,omitempty"`
+	// If true enable reloading via HTTP
+	HotReload *bool `json:"hotReload,omitempty"`
```

Comment:
How to utilize this flag if enabled?
In which version it's added to fluentbit?
@wanjunlei pls take a look

Reply:
On how to use this flag can see https://docs.fluentbit.io/manual/administration/hot-reload
The version added to fluentbit should be v2.1 https://fluentbit.io/announements/v2.1.0/

Reply:
We've already add `--enable-hot-reload` flag in fluentbit watch:
https://github.com/fluent/fluent-operator/blob/master/cmd/fluent-watcher/fluentbit/main.go#L76C20-L76C39

And the config will be reload whenever there is a config change by sending SIGHUP signal:
https://github.com/fluent/fluent-operator/blob/master/cmd/fluent-watcher/fluentbit/main.go#L126

FluentBit doc says `Fluent Bit supports the hot reloading feature when enabled via the configuration file or command line with -Y or --enable-hot-reload option.` which means it's not necessary to add it to configfiles because the command line arg already works.

And fluent operator already supports HTTP settings https://github.com/fluent/fluent-operator/pull/1286/files#diff-7aad9bf9eeeaa93f3a61c5ea93ea5a30d57bcfa6307ec4ea188222fdf69f438cR99-R105

![image](https://github.com/user-attachments/assets/39600c17-4342-445d-a675-15a25a630578)


@wanjunlei what do you think? do we need to add the hot reload flag to configfiles?

Reply:
fluentbit already supports hot reload via the `--enable-hot-reload` flag, so there is no need to add `Hot_Reload` to theconfig file.

Reply:
There are several main reasons for adding this parameter:
1. The reason for debugging. Because fluent-operator wraps the watcher layer, you can only restart the process by restarting the container or changing the configuration of the CR when debugging, restarting the container will be more troublesome and changing the configuration is not easy to locate the problem, so we need a way to send a signal to restart manually for hot loading.
2. This parameter is one of the officially recommended ways to use this can also be harmonized with the official.
3. Historical legacy issues, before the signal for hot loading in the watcher layer has been changed, due to the fluentbit version upgrade, so the signal to trigger hot loading has also been changed, and the way through the http is compatible with the lower version, but also for the sake of unity.
4. Compared to the practice of adding parameters in the watcher layer, I think it is also a simple way to configure directly in the configuration file.

Reply:
You have nicely explained this!